nlp/optimize/tron.py

```diff
@@ -362,7 +370,7 @@ def projected_linesearch(self, x, l, u, g, d, H, alpha=1.0):
 
         # Compute the final iterate and step.
         s = projected_step(x, alpha * d, l, u)
-        x = project(x + alpha * s, l, u)
+        x = project(x + s, l, u)
```

Comment:
I know there was a bug. It's not that I prefer one or the other. They're two different things.